### ISSUE TITLE
Speed rolls from MidiQoL don't work with TAH

### ISSUE THREAD

**Issue by Bogus73** [Drental/fvtt-tokenactionhud#151]
Hi, 
as MidiQoL is using the default keybindings now they do not work with the Actionhud anylonger.
They work just fine in the character sheet but if I press the key for rolling with adv/disadv it sometimes doesn't roll or doesn't stop rolling with adv/disadv even if I don't press the respective key.
I am not sure what the problem is but wanted to make you aware of the issue as I love both modules a lot and my players finally use it.

Thx for your work!

**Comment by Drental** [comment]
should be fixed by a midi update.

**Comment by Bogus73** [comment]
Tried it and with updated midi, readme said it would fix it but unfortunately it doesn't.
It works perfectly well in the character sheet just not from the hud.

**Comment by tposney** [comment]
Short version: change line 99-101 in tokenactionhud.js from
```
      if (target.tagName !== "BUTTON") target = e.currentTarget.children[0];

      let value = target.value;
 ```
 to
```
       if (target.tagName !== "BUTTON") target = e.currentTarget.children[0];
       e.blur();
      let value = target.value;
```
and the problem goes away.

I think I know what is going on.
When you click on a button in TAH the BUTTON gets focus, then foundry throws away the release of the key so adv/dis are stuck. The button keeps focus until you move away from TAH.  So even without midi the ALT/CTRL keys get stuck if you clcik in TAH with those keys pressed.

It just shows up in midi more often.

A workaround until the fix is to move the mouse away from TAH before releasing the accelerator key.

**Comment by tposney** [comment]
tl;dr 0.9.05 of midi will attempt to work around the stuck keys problem even without the suggested fix.   

It turns out that this is not just an issue with TAH - any app that has input fields (like buttons) that grab focus when you click will leave the modifier keys "stuck" on if the key was pressed before the field got focus. This includes TAH, the dnd5e character sheet and so on.

It's not clear to me that the current foundry behaviour is what one would expect, so I've coded a work around for it in midi-qol and raised an issue on the foundry gitlab.

**Comment by Drental** [comment]
> Short version: change line 99-101 in tokenactionhud.js from
> 
> ```
>       if (target.tagName !== "BUTTON") target = e.currentTarget.children[0];
> 
>       let value = target.value;
> ```
> 
> to
> 
> ```
>        if (target.tagName !== "BUTTON") target = e.currentTarget.children[0];
>        e.blur();
>       let value = target.value;
> ```
> 
> and the problem goes away.
> 
> I think I know what is going on. When you click on a button in TAH the BUTTON gets focus, then foundry throws away the release of the key so adv/dis are stuck. The button keeps focus until you move away from TAH. So even without midi the ALT/CTRL keys get stuck if you clcik in TAH with those keys pressed.
> 
> It just shows up in midi more often.
> 
> A workaround until the fix is to move the mouse away from TAH before releasing the accelerator key.

well, that didn't work out 😓

**Comment by thatlonelybugbear** [comment]
@Drental and @tposney.

When tposney was trying to find a fix and he came up with the `e.blur()` solution, we tested on my server and it was working OK after some quick testing.

The thing is that I added that in the minified, `bundle.min.js` in my token-action-hud module folder, like so:

`{activateListeners(t){const e=".tah-action",handleClick=t=>{let e=t.target;"BUTTON"!==e.tagName&&(e=t.currentTarget.children[0]);let i=e.value;e.blur();try{this.rollHandler.handleActionEvent(t,i)}`

Do you think that it would be needing a different syntax when you add the line in the `tokenactionhud.js`. I have honestly no idea, if that makes a difference or if another issue would be still present that I haven't encountered. I know that Midi kind of rolled back on the way it was handling the keybinds in the newest version.

Haven't tried the newest midi version yet with Token Action HUD as I am having a session soon, so I am still on midi 0.9.04 and TAH v2.0.19 with the line `e.blur();` added and I guess it works 🤷

**Comment by Drental** [comment]
merged and tested, thanks
